base/deprecated.jl

```diff
@@ -2795,6 +2583,31 @@ end
 @deprecate_moved sum_kbn "KahanSummation"
 @deprecate_moved cumsum_kbn "KahanSummation"
 
+# PR #25249: SparseArrays to stdlib
+## the Base.SparseArrays module itself and exported types are deprecated in base/sysimg.jl
+## functions that were re-exported from Base
+@deprecate_moved nonzeros   "SparseArrays" true true
+@deprecate_moved permute    "SparseArrays" true true
+@deprecate_moved blkdiag    "SparseArrays" true true
+@deprecate_moved dropzeros  "SparseArrays" true true
+@deprecate_moved dropzeros! "SparseArrays" true true
+@deprecate_moved issparse   "SparseArrays" true true
+@deprecate_moved sparse     "SparseArrays" true true
+@deprecate_moved sparsevec  "SparseArrays" true true
+@deprecate_moved spdiagm    "SparseArrays" true true
+@deprecate_moved sprand     "SparseArrays" true true
+@deprecate_moved sprandn    "SparseArrays" true true
+@deprecate_moved spzeros    "SparseArrays" true true
+@deprecate_moved rowvals    "SparseArrays" true true
+@deprecate_moved nzrange    "SparseArrays" true true
+@deprecate_moved nnz        "SparseArrays" true true
+## functions that where exported from Base.SparseArrays but not from Base
+@deprecate_moved droptol!   "SparseArrays" false true
+## deprecated functions that are moved to stdlib/SparseArrays/src/deprecated.jl
+@deprecate_moved spones     "SparseArrays" true true
+@deprecate_moved speye      "SparseArrays" true true
```

Comment:
Calling `spones` and `speye` now will result in users first getting redirected to `SparseArrays`, and then get a message that those methods are deprecated. But perhaps no big deal.